packages/playwright-ct-react17/registerSource.mjs

```diff
@@ -36,9 +36,11 @@ export function register(components) {
 
 /**
  * @param {Component} component
- * @returns {JSX.Element}
  */
 function render(component) {
+  if (typeof component !== 'object' || Array.isArray(component))
+    return component;
+
```

Comment:
This is the same check as in react18: https://github.com/microsoft/playwright/blob/main/packages/playwright-ct-react/registerSource.mjs#L43